docs/component_authors/networking.md

```diff
@@ -0,0 +1,122 @@
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+  egress:
+  - {}
+```
+
+To configure the UAA `Pod` to allow ingress traffic from your new component, you would add an additional element to the `from` array. Something like this:
+
+```
+- podSelector:
+    matchLabels:
+      app.kubernetes.io/name: my-new-component
+  namespaceSelector:
+    matchLabels:
+      cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
+```
+
+This configures the UAA `NetworkPolicy` to allow ingress traffic from any `Pod` that has the label `app.kubernetes.io/name=my-new-component` in a namespace with the `cf-for-k8s.cloudfoundry.org/cf-system-ns=""` label.
+
+Once you've tested it and everything works, make a PR to update the `NetworkPolicy` in `config/network-policies.yaml`!
+
+### Namespaces and Network Policies
+`NetworkPolicies` apply to `Pods` within a **single namespace**.
+
+If you're creating a new namespace we highly recommend creating an initial ["default deny all"](https://kubernetes.io/docs/concepts/services-networking/network-policies/#default-deny-all-ingress-and-all-egress-traffic) `NetworkPolicy` to block all ingress and egress traffic to `Pods` in the namespace.
+
+```yaml
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+  namespace: YOUR_NAMESPACE_NAME
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+Then add in the appropriate `allow` rules through additional `NetworkPolicies`.
+
+
+## Istio Sidecar Injection
+
+### Namespaces and Istio Sidecar Injection
+CF for Kubernetes uses Istio's [automatic mTLS](https://istio.io/latest/docs/tasks/security/authentication/authn-policy/#auto-mutual-tls) with a `STRICT` authentication policy. This means that all `Pods` that have an Istio sidecar will require any `Pods` that communicate with them to also have an Istio sidecar and use mutual TLS to secure traffic. Traffic that doesn't use mTLS or traffic that does not come from an Istio sidecar will be rejected and the connection will fail.
+
+When you create a new namespace, in order to take advantage of all of Istio's features, and communicate with other `Pods` that Istio sidecars, your namespace must have an `istio-injection=enabled` label to turn on automatic [Sidecar Injection](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/). This makes it so the pods in your namespace will contain an [Istio sidecar proxy](https://istio.io/latest/docs/reference/config/networking/sidecar/) running alongside them. In CF for Kubernetes, Istio sidecar injection is added automatically by a `ytt` [overlay](https://github.com/cloudfoundry/cf-for-k8s/blob/master/config/networking.yml#L63).
+
+To confirm your namespace has automatic sidecar injection enabled, you can describe your namespace with `kubectl get namespace <namespace_name> -L istio-injection` and see that Istio Injection is set to "enabled".
+
```

Comment:
It appears the istio components living in the `istio-system` namespace have a metrics endpoint at ":15014/metrics". However, we intend to deploy Prometheus into the `cf-system` namespace which has the sidecar injection enabled, whereas `istio-system` does not. How would we be able to scrape the istio metrics?

Reply:
I don't think it should be a problem. Both Pods must have sidecars if the "server" Pod has an Istio sidecar. The client Pod should be able to talk to other things that don't have one (based on how we've configured it at least).

[client sidecar] ----------any protocol---------> [something *without* sidecar] ✅
[client sidecar] ---------mTLS----------> [something with sidecar] ✅
[client *without* sidecar] ---------any protocol----------> [something with sidecar] 🛑

Reply:
We'll try and take another pass at this section of the docs to clarify this.

Reply:
So you're saying because istio components don't require it, the Prometheus pod should be able to connect regardless?

I added the network policy below and added annotations to istio-telemetry but am getting this error when Prometheus tries to scrape:  `server returned HTTP status 503 Service Unavailable`.

```
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: allow-prometheus-scrape
  namespace: istio-system
spec:
  podSelector: {}
  policyTypes:
  - Ingress
  ingress:
  - from:
    - namespaceSelector:
        matchLabels:
          cf-for-k8s.cloudfoundry.org/cf-system-ns: ""
      podSelector:
        matchLabels:
          what-am-i: prometheus
```

Reply:
Do you also have to add a network policy that allows egress from the prometheus pod to that ns?

Reply:
Played around with this more and was actually able to see metrics for some of the istio-components using the above network policy. However only half of the istio components worked (citadel, pilot, sidecar-injector) and the other half fails (galley, telemetry, policy). I can open this as an issue directly on cf-k8s-networking though.

Reply:
Hmm, a lot of those components are being consolidated as part of the Istio 1.6.x upgrade (we'll end up with just two processes `istiod` and `istio-ingressgateway`) so this might change a bit anyway... but I think if you could create the issue that would help us keep it on the radar!

Reply:
We attempted to make that part of the docs more clear. Let us know if you have any other suggestions!
https://github.com/cloudfoundry/cf-for-k8s/pull/270/commits/8d3fd6e6d2142f0f58855f5b89d509437d26cde1